redbot/cogs/permissions/converters.py

```diff
@@ -37,26 +38,33 @@ async def convert(
             if user is not None:
                 return user
 
-            for guild in bot.guilds:
+            async for guild in AsyncGen(bot.guilds, steps=100):
                 role: discord.Role = guild.get_role(_id)
                 if role is not None:
                     return role
-
+        all_roles = [
+            roles async for guild in AsyncGen(bot.guilds, steps=100) for roles in guild.roles
+        ]
         objects = itertools.chain(
             bot.get_all_channels(),
             bot.users,
             bot.guilds,
-            *(filter(lambda r: not r.is_default(), guild.roles) for guild in bot.guilds),
+            *(
+                filter(lambda r: not r.is_default(), roles)
+                async for roles in AsyncGen(all_roles, steps=100)
+            ),
```

Comment:
This and the `all_roles` list that was added above can be simplified to:
```suggestion
            *(
                filter(lambda r: not r.is_default(), guild.roles)
                async for guild in AsyncGen(bot.guilds, steps=100)
            ),
```
That also means that `all_roles` from above can be removed.